### PR TITLE
[BEAM-5636] Java support for custom dataflow worker jar

### PR DESCRIPTION
**Pull request by boyuanzz**
Added pipeline option: dataflowWorkerJar=${PATH} and experimental flag: use_staged_dataflow_worker_jar to make java pipeline can be launched with custom dataflow worker jar.

R: @herohde 
cc: @HuangLED

### CODE REVIEW

**Comment by pabloem** [comment]
Fwiw you can run `./gradlew :..project..:check` to run unit tests and lint checks without running dataflow jobs, which should be useful ; )

**Comment by boyuanzz** [comment]
Re: @pabloem Thanks! Working on minor changes now.

File: sdks/java/container/boot.go [hunk 1]
```diff
@@ -104,6 +104,9 @@ func main() {
 		filepath.Join(jarsDir, "beam-sdks-java-harness.jar"),
 	}
 	for _, md := range artifacts {
+		if strings.HasPrefix(md.Name, "beam-runners-google-cloud-dataflow-java-fn-api-worker") {
```
Comment (herohde) [inline sdks/java/container/boot.go]: We should only make this check if the experiment is set. Also, the name will change to "dataflow-worker.jar" when the artifact bug is fixed.

Comment (boyuanzz) [reply sdks/java/container/boot.go]: The purpose here is, if there is java worker jar in artifacts, then this jar should not be included into sdk harness classpath, which seems like we don't need to check experiment. wdyt?

**Comment by boyuanzz** [comment]
Re: @herohde Please take another look at this. Thanks~

File: sdks/java/container/boot.go [hunk 2]
```diff
@@ -103,7 +103,17 @@ func main() {
 		filepath.Join(jarsDir, "slf4j-jdk14.jar"),
 		filepath.Join(jarsDir, "beam-sdks-java-harness.jar"),
 	}
+
+  var has_worker_experiment = strings.Contains(options, "use_staged_dataflow_worker_jar")
```
Comment (herohde) [inline sdks/java/container/boot.go]: nit: go uses camelCase for local variables. You also need to run "gofmt -w ." to fix the indentation.

Comment (boyuanzz) [reply sdks/java/container/boot.go]: Fixed, thanks~

File: sdks/java/container/boot.go [hunk 3]
```diff
@@ -103,7 +103,17 @@ func main() {
...
 		filepath.Join(jarsDir, "beam-sdks-java-harness.jar"),
 	}
+
+	var hasWorkerExperiment = strings.Contains(options, "use_staged_dataflow_worker_jar")
 	for _, md := range artifacts {
+		if hasWorkerExperiment {
+			if strings.HasPrefix(md.Name, "beam-runners-google-cloud-dataflow-java-fn-api-worker") {
+				continue
+			}
+			if strings.HasPrefix(md.Name, "dataflow-worker.jar") {
```
Comment (herohde) [inline sdks/java/container/boot.go]: Small comment: this can be == instead.

**Comment by herohde** [comment]
@boyuanzz Btw, please also update the doc describing this change to include Java.

**Comment by boyuanzz** [comment]
@herohde this PR is ready to merge.

**Pull request merged**